src/DataStructures/Tensor/EagerMath/CartesianToSpherical.cpp

```diff
@@ -5,6 +5,7 @@
 
 #include <cstddef>
 
+#include "DataStructures/DataVector.hpp"
```

Comment:
Good question! I think practically speaking the answer is "no." Here are a list of things I can immediately think of that I think would be challenging to overcome (in no particular order):
1. We would need to figure out how to have CMake use a different compiler for different libraries.
2. We would need to filter out the compiler arguments that Kokkos adds for nvcc that GCC/Clang can't handle. I'm already running into this with our linker wrapper, but I haven't dug into what's going on there.
3. There may be ABI incompatibilities now or in the future because things like `__host__` and `__device__` may or may not be included in the function signature. This is implementation defined and Clang and nvcc currently behave differently.
4. We need essentially all header files to be compatible with nvcc anyway since they get included in PointwiseFunctions, NumericalAlgorithms, ParallelAlgorithms, and the final executable (I think we need to compile the final executable with nvcc, I don't see a way around that). This means we would only benefit from not having to make the cpp files compatible, which is definitely not negligible, but I worry that this will be very confusing in the long term. I've found that it is also easier to make one library at a time work with nvcc rather than trying to deal with all the possible headers files when building at the end. Essentially, incremental conversion, even of full libraries, seems easier than many libraries all at once from my experience so far.

Reply:
Looks like CMake supports separate host and device compilers with [CUDA_SEPARABLE_COMPILATION](https://cmake.org/cmake/help/latest/prop_tgt/CUDA_SEPARABLE_COMPILATION.html). I haven't seen support for this in Kokkos though.

We can also try `-DCMAKE_CUDA_COMPILER=clang++` to avoid having to keep compatibility with `nvcc` for the whole code.

Reply:
That might be worth trying. I just don't know how it would work with Kokkos because kokkos actually has a wrapper around the Nvidia compiler.